frontend/src/metabase/nav/components/ProfileLink/ProfileLink.unit.spec.tsx

```diff
@@ -0,0 +1,192 @@
+
+      openMenu();
+
+      REGULAR_ITEMS.forEach(title => {
+        expect(screen.getByText(title)).toBeInTheDocument();
+      });
+      expect(screen.queryByText("Admin settings")).not.toBeInTheDocument();
+    });
+
+    it("should show the proper set of items for admin users", () => {
+      setupHosted({ isAdmin: true });
+
+      openMenu();
+
+      HOSTED_ITEMS.forEach(title => {
+        expect(screen.getByText(title)).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe("help link", () => {
+    describe("when the setting is set to hidden", () => {
+      it("should return not be visible", () => {
+        setup({
+          helpLinkSetting: "hidden",
+        });
+        openMenu();
+
+        const link = screen.queryByRole("link", { name: /help/i });
+
+        expect(link).not.toBeInTheDocument();
+      });
+    });
+
+    describe("when the setting is a custom url", () => {
+      it("should return  the custom destination", () => {
+        setup({
+          helpLinkSetting: "https://custom.example.org/help",
+        });
+        openMenu();
+
+        const link = screen.getByRole("link", { name: /help/i });
+
+        expect(link).toBeInTheDocument();
+        expect(link).toHaveProperty("href", "https://custom.example.org/help");
+      });
+    });
+
+    describe("when the setting is `metabase_default`", () => {
+      describe("when admin on paid plan", () => {
+        it("should return the default /help-premium link", async () => {
+          setup({
+            isAdmin: true,
+            isPaidPlan: true,
+            helpLinkSetting: "metabase_default",
+          });
+          openMenu();
+          const link = screen.getByRole("link", { name: /help/i });
+
+          expect(link).toBeInTheDocument();
+          const mockBugReportDetails = encodeURIComponent(
+            JSON.stringify("mockBugReportDetails"),
+          );
+          const expected = `https://www.metabase.com/help-premium?utm_source=in-product&utm_medium=menu&utm_campaign=help&instance_version=v1&diag=${mockBugReportDetails}`;
+          await waitFor(() => expect(link).toHaveProperty("href", expected));
+
+          expect(link).toHaveProperty("href", expected);
+        });
+      });
+
+      describe("when non admin", () => {
+        it("should return the default /help link", () => {
+          setup({
+            isAdmin: false,
+            isPaidPlan: true,
+            helpLinkSetting: "metabase_default",
+          });
+          openMenu();
+          const link = screen.getByRole("link", { name: /help/i });
+
+          expect(link).toBeInTheDocument();
+          expect(link).toHaveProperty(
+            "href",
+            "https://www.metabase.com/help?utm_source=in-product&utm_medium=menu&utm_campaign=help&instance_version=v1",
+          );
+        });
+      });
+    });
+  });
+});
+
+const openMenu = () =>
+  fireEvent.click(screen.getByRole("img", { name: /gear/i }));
```

Comment:
While we don't deprecate `fireEvent`, we encourage using `userEvent` which emulates users' event so they're more realistic.